pkg/operation/cloudbotanist/types.go

```diff
@@ -35,6 +35,7 @@ type CloudBotanist interface {
 	GenerateCloudProviderConfig() (string, error)
 	RefreshCloudProviderConfig(map[string]string) map[string]string
 	GenerateCloudConfigUserDataConfig() *common.CloudConfigUserDataConfig
+	GenerateETCDStorageClassConfig() map[string]interface{}
```

Comment:
Can you elaborate on how this will look like with extensibility?

From https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md#reconciliation-flow:

> Gardener does only deploy the etcd StatefulSet without backup-restore sidecar at all.The cloud-specific Botanist registers a Mutating Webhook and adds the backup-restore sidecar, and it also creates the Secret needed by the backup-restore sidecar.

This cannot be done this way anymore as the etcd deployment itself does now require a special storage class before it can be deployed.

Reply:
As discussed via phone: The extension controllers would create the storage class after they start up. When Gardener creates the etcd Statefulset they will inject the sidecar and modify the volume claim templates accordingly.